users/replicaJunction/process_records.c

```diff
@@ -0,0 +1,140 @@
+#include QMK_KEYBOARD_H
+#include "process_records.h"
+
+uint8_t mod_state;
+
+
+__attribute__ ((weak))
+bool process_record_user_kb(uint16_t keycode, keyrecord_t *record) {
+    return true;
+}
+
+// Runs for each key down or up event.
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    // Returning true here will cause QMK to continue handling the key normally.
+    // Returning false indicates that we've handled everything the keycode should do, and QMK
+    // should not continue handling the keypress.
+    //
+    // NOTE: There is also a process_record_kb function that can be defined in the keyboard-
+    // specific code. This allows the keyboard to have its own process_record function.
+    // This is supposed to be "higher" than the user function, meaning the kb function
+    // is shared for all keymaps for the keyboard.
+    //
+    // For this reason, I add my own function, called process_record_user_kb, and at the end
+    // of this function, I defer to that one if it exists.
+    // return process_record_user(keycode, record);
+
+
+    // Custom keycode / function handling, based on the core function
+    // process_record_quantum
+    // https://github.com/qmk/qmk_firmware/blob/master/quantum/quantum.c
+
+    if (!(
+#ifdef USER_CAPS_WORD_ENABLE
+        process_record_caps_word(keycode, record) &&
+#endif
+#ifdef USER_MOUSE_JIGGLE_ENABLE
+        process_record_mouse_jiggle(keycode, record) &&
+#endif
+#ifdef USER_NUM_WORD_ENABLE
+        process_record_num_word(keycode, record) &&
+#endif
+#ifdef USER_SECRETS_ENABLE
+        process_record_secrets(keycode, record) &&
+#endif
+#ifdef USER_SUPER_ALT_TAB_ENABLE
+        process_record_super_alt_tab(keycode, record) &&
+#endif
+        true)) {
+        return false;
+    }
+
+
+    // Miscellaneous keycode handling
+    mod_state = get_mods();
+
+    switch(keycode)
+    {
+        case QK_MAKE: {
+            if (record->event.pressed)
+                SEND_STRING("qmk compile --keyboard " QMK_KEYBOARD " --keymap " QMK_KEYMAP);
+            return false;
+        }
+        case QK_FLSH: {
+            if (record->event.pressed) {
+                SEND_STRING("qmk flash --keyboard " QMK_KEYBOARD " --keymap " QMK_KEYMAP
+                #if defined(BOOTLOADER_CATERINA)
+                    " --bootloader avrdude"
+                #elif (defined(BOOTLOADER_DFU) || defined(BOOTLOADER_LUFA_DFU) || defined(BOOTLOADER_QMK_DFU))
+                    " --bootloader dfu"
+                #endif
+                );
```

Comment:
```suggestion
                SEND_STRING("qmk flash --keyboard " QMK_KEYBOARD " --keymap " QMK_KEYMAP);
```
Bootloader stuff shouldn't be needed, `qmk flash` will run the correct flashing tool based on what's in the `rules.mk`.